nodejs/tfwconnector/TWFConnector.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+const zmq = require('zeromq'),
+    subSock = zmq.socket('sub'),
+    pushSock = zmq.socket('push');
+
+const subPort = process.env.TFW_PUBLISHER_PORT || 7654,
+    pushPort = process.env.TFW_RECEIVER_PORT || 8765;
+
+const subAddress = "tcp://localhost:" + subPort,
+      pushAddress = "tcp://localhost:" + pushPort;
+
+
+class TFWConnector {
+    constructor() {
+        this.pushSocket = pushSock.connect(pushAddress);
+        this.subSocket = subSock.connect(subAddress);
+        console.log("Push socket connected to: " + pushAddress + "\n" + "Subscribe socket connected to: " + subAddress);
+    }
+
+    subscribe(key) {
+        this.subSocket.subscribe(key);
+    }
+
+    sendMessage(message, name = "avataobot") {
+        let key = "message";
+        let data = {
+            key: key,
+            data: {
+                originator: name,
+                timestamp: Date.now(),
+                message: message
+            },
+            trigger: ''
+        };
+        let messageToSend = createTFWMessage(key, data);
+        this.pushSocket.send(messageToSend);
+    }
+
+    setView(viewType) {
```

Comment:
These convenience wrappers and nice, but they should be distributed in a separate file to emphasize that they are fully optional and should not be depended on (i.e. a `TerminalEventHandler` on the backend is not available in all challenges).